Submitty/Submitty
# [Refactor:Router] New URL for MiscController

### URL Changes

URLs may need some adjustment as MiscController's usage is everywhere. Some of them need course information, which make things more complicated.

- GET `/server_time`
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/encode_pdf`
- GET `/{_semester}/{_course}/display_file`
- GET `/read_file`
- GET `/{_semester}/{_course}/download`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/download_zip`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/download_zip`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/bulk/progress"`

### Testing
Manually tested the following:
- Get server time
- See embedded pdf in the grading page
- download course material
- download students' zip file
- Display pdf and json
- Get bulk info
